File: scripts/extract_mermaid.ts
#!/usr/bin/env bun
/**
 * Extract the mermaid block from a .mmd file (strips YAML front matter).
 *
 * Usage:
 *   bun run scripts/extract_mermaid.ts <input.mmd> <output.mmd>
 *   bun scripts/extract_mermaid.ts input.mmd output.mmd
 *
 * Example:
 *   bun scripts/extract_mermaid.ts docs/diagrams/src/system-overview.mmd /tmp/out.mmd
 */

import { readFileSync, writeFileSync } from "fs";

const [, , inputPath, outputPath] = process.argv;

if (!inputPath || !outputPath) {
  console.error(`Usage: ${process.argv[1]} <input.mmd> <output.mmd>`);
  process.exit(1);
}

const content = readFileSync(inputPath, "utf8");

const match = content.match(/\x60\x60\x60mermaid\n([\s\S]*?)\n\x60\x60\x60/);
if (!match) {
  console.error(`No mermaid block found in ${inputPath}`);
  process.exit(1);
}

writeFileSync(outputPath, match[1]);

File: scripts/extract_mermaid.py
#!/usr/bin/env python3
"""Extract mermaid block from a .mmd file (strips YAML front matter)."""
import re, sys

if len(sys.argv) != 3:
    print(f"Usage: {sys.argv[0]} <input.mmd> <output.mmd>", file=sys.stderr)
    sys.exit(1)

input_path, output_path = sys.argv[1], sys.argv[2]

with open(input_path) as f:
    content = f.read()

match = re.search(r'\x60\x60\x60mermaid\n(.*?)\n\x60\x60\x60', content, re.DOTALL)
if not match:
    print(f"No mermaid block in {input_path}", file=sys.stderr)
    sys.exit(1)

with open(output_path, 'w') as out:
    out.write(match.group(1))